be/src/column/struct_column.cpp

```diff
@@ -223,55 +219,61 @@ void StructColumn::append_default(size_t count) {
 }
 
 uint32_t StructColumn::serialize(size_t idx, uint8_t* pos) {
-    // TODO(SmithCruise) Not tested.
     uint32_t ser_size = 0;
     for (ColumnPtr& column : _fields) {
-        ser_size += column->serialize(idx, pos);
-    }
-    for (size_t i = 0; i < _field_names->size(); i++) {
-        ser_size += _field_names->serialize(i, pos);
+        ser_size += column->serialize(idx, pos + ser_size);
     }
     return ser_size;
 }
 
 uint32_t StructColumn::serialize_default(uint8_t* pos) {
-    // TODO(SmithCruise) Not tested.
     uint32_t ser_size = 0;
     for (ColumnPtr& column : _fields) {
-        ser_size += column->serialize_default(pos);
-    }
-    for (size_t i = 0; i < _field_names->size(); i++) {
-        ser_size += _field_names->serialize(i, pos);
+        ser_size += column->serialize_default(pos + ser_size);
     }
     return ser_size;
 }
 
 void StructColumn::serialize_batch(uint8_t* dst, Buffer<uint32_t>& slice_sizes, size_t chunk_size,
                                    uint32_t max_one_row_size) {
-    // TODO(SmithCruise) Not tested.
     for (size_t i = 0; i < chunk_size; ++i) {
         slice_sizes[i] += serialize(i, dst + i * max_one_row_size + slice_sizes[i]);
     }
 }
 
 const uint8_t* StructColumn::deserialize_and_append(const uint8_t* pos) {
-    DCHECK(false) << "Dont support it";
-    return nullptr;
+    for (size_t i = 0; i < _fields.size(); i++) {
+        pos = _fields[i]->deserialize_and_append(pos);
+    }
+    return pos;
 }
 
 void StructColumn::deserialize_and_append_batch(Buffer<Slice>& srcs, size_t chunk_size) {
-    DCHECK(false) << "Dont support it";
+    reserve(chunk_size);
+    for (size_t i = 0; i < chunk_size; ++i) {
+        srcs[i].data = (char*)deserialize_and_append((uint8_t*)srcs[i].data);
+    }
+}
+
+uint32_t StructColumn::max_one_element_serialize_size() const {
+    // TODO: performance optimization.
+    size_t n = size();
+    uint32_t max_size = 0;
+    for (size_t i = 0; i < n; i++) {
+        uint32_t row_max_size = 0;
+        for (const auto& column : _fields) {
+            row_max_size += column->max_one_element_serialize_size();
+        }
+        max_size = std::max(max_size, row_max_size);
```

Comment:
I think in this code the row_max_size is the sum size of max_element in each fields, so the the outer loop will get the same row_max_size.

Reply:
fixed